cpp/src/groupby/sort/group_reductions.hpp

```diff
@@ -42,6 +42,21 @@ std::unique_ptr<column> group_sum(column_view const& values,
                                   rmm::cuda_stream_view stream,
                                   rmm::mr::device_memory_resource* mr);
 
+/**
+ * @brief Internal API to calculate groupwise product
+ *
+ * @param values Grouped values to get product of
+ * @param num_groups Number of groups
+ * @param group_labels ID of group that the corresponding value belongs to
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ */
+std::unique_ptr<column> group_product(column_view const& values,
```

Comment:
Can you add more description for the API, please? Adding one simple example would be great. I had trouble understanding the unfamiliar APIs in cudf with such short descriptions and had to read the entire of their implementation to know what they are doing.

Reply:
These are Internal APIs. These are used only in groupby. Their usage is usually at only one place `aggregate.cpp` or `scan.cpp`.
Do you still prefer to have a usage example?

Reply:
I mean, a more detailed description not only helps public users but also helps other people (other than you) to understand it when they need to use it in the future in their development.